### DISCUSSION TITLE
Help configuring leap.nvim and mini.surround

### DISCUSSION THREAD

**Discussion by albertfgu** [LazyVim/LazyVim#619] [Q&A]
I'm having trouble understanding the LazyVim configuration for these plugins and how to remap them to my desired bindings. **My overall goal is to put `mini.surround` bindings on `s` (instead of `gz`, which LazyVim remaps them to) and to put `leap.nvim`on `gs/gS`**

Here is the leap configuration from https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/editor.lua#L197:
```
    "ggandor/leap.nvim",
    keys = {
      { "s", mode = { "n", "x", "o" }, desc = "Leap forward to" },
      { "S", mode = { "n", "x", "o" }, desc = "Leap backward to" },
      { "gs", mode = { "n", "x", "o" }, desc = "Leap from windows" },
    },
    config = function(_, opts)
      local leap = require("leap")
      for k, v in pairs(opts) do
        leap.opts[k] = v
      end
      leap.add_default_mappings(true)
      vim.keymap.del({ "x", "o" }, "x")
      vim.keymap.del({ "x", "o" }, "X")
    end,
  },
```

1. I'm having a hard time understanding this, as it looks quite different from other lazy.nvim configurations. Looking at the source code for leap isn't helping as I don't see where it's using the user options (https://github.com/ggandor/leap.nvim/blob/main/lua/leap/user.lua#L1). **How can I create a config to overwrite the keybindings from s/S to gs/gS?**

2. I tried creating a file `lua/plugins/editor.lua` and overrode the bindings defined at https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/coding.lua#L123.
```
return {
  "echasnovski/mini.surround",
  opts = {
    mappings = {
      add = "sa", -- Add surrounding in Normal and Visual modes
      delete = "sd", -- Delete surrounding
      find = "sf", -- Find surrounding (to the right)
      find_left = "sF", -- Find surrounding (to the left)
      highlight = "sh", -- Highlight surrounding
      replace = "sr", -- Replace surrounding
      update_n_lines = "sn", -- Update `n_lines`
    },
  },
  config = function(_, opts)
    -- use gz mappings instead of s to prevent conflict with leap
    require("mini.surround").setup(opts)
  end,
}
```
**Q: is there a better way to do this than manually copy-pasting and redefining the whole table? e.g. is there a shorter way to say "undo the settings set by LazyVim and use defaults from mini.surround"**

3. After doing this, I'm having some weird keymap conflicts where pressing `sa` would go into mini.surround correctly, but if I type `s` and pause it will go into Leap. Why is this happening?

4. **The previous issue remains even after disabling leap entirely by adding ` { "ggandor/leap.nvim", enable = false }` to the config** where pausing after `s` actives Leap

5. Finally, assuming everything else works, would I have to manually overwrite the which-key entries for `gz` and `s`? https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/editor.lua#L224

**Comment by folke** [comment]
`enable` => `enabled`

If you don't want to use the default leap bindings, then you need to remove the line `leap.add_default_mappings(true)` and manually set the keymaps.

And yes, you'll have to override the which-key defaults as well.

**Comment by albertfgu** [comment]
Oops, thanks!

So does this look right for a minimal config that drops leap.nvim entirely and restores the defaults of surround? In particular, there's no way to say "undo the mappings from LazyVim" (i.e. restore default surround settings) right?
```
return {
	{ "ggandor/leap.nvim", enabled = false },
	{
		"echasnovski/mini.surround",
		opts = {
			mappings = {
				add = "sa", -- Add surrounding in Normal and Visual modes
				delete = "sd", -- Delete surrounding
				find = "sf", -- Find surrounding (to the right)
				find_left = "sF", -- Find surrounding (to the left)
				highlight = "sh", -- Highlight surroundng
				replace = "sr", -- Replace surroundng
				update_n_lines = "sn", -- Update `n_lines`
			},
		},
	},
	{
		"folke/which-key.nvim",
		opts = {
			defaults = {
				["gz"] = nil,
				["s"] = { name = "+surround" },
				["<leader>f"] = { name = "+file/find" }, -- remove find?
			},
		},
	},
}
```
Also the which-key isn't working here, I haven't yet figured out how to remove something from a table.
